### ISSUE TITLE
Conflicting types for 'localtime'

### ISSUE THREAD

**Issue by nicholas** [keenlabs/KeenClient-iOS#111]
Using the Cocoapods version of the KeenClient (3.4.1) with Xcode 6.4 (iOS 8.4) is producing the following errors in keen_io_sqlite3.c:

1) .../Pods/KeenClient/Library/sqlite-amalgamation/keen_io_sqlite3.c:14983:8: Conflicting types for 'localtime' ![screenshot 2015-07-22 17 20 05](https://cloud.githubusercontent.com/assets/52206/8838924/4fe956e0-3096-11e5-8c33-b144470e0133.png)

2) .../Pods/KeenClient/Library/sqlite-amalgamation/keen_io_sqlite3.c:15021:13: Definition of 'struct tm' must be imported from module 'Darwin.C.time' before it is required ![screenshot 2015-07-22 17 20 16](https://cloud.githubusercontent.com/assets/52206/8838923/4fe7c3ca-3096-11e5-83df-ce424ccad10d.png)

I'm going to swap over to the binary installation method tomorrow to see if that helps, but thought this was worth pointing out.

**Comment by heitortsergent** [comment]
@nicholas sorry about this error. :(

Did changing over to the binary installation fix it? Do you remember which version of the library you were using?

I just created a blank project and installed the latest version of the library via CocoaPods and don't see any errors.

**Comment by nicholas** [comment]
Oh no worries. The switch did indeed fix the issue (and we're loving Keen btw!).

I'll give it another shot by tomorrow and get you more details on versions and whatnot.

**Comment by heitortsergent** [comment]
@nicholas thanks a lot, really appreciate that!

And I'm glad you're loving Keen! :dancer:

**Comment by nicholas** [comment]
Ok, yep, still able to reproduce it. However, not with a brand new project. So it looks like it's something on my end... I'm guessing a conflict with another pod. It's going to take me a bit to dig into it to find out which one, but I'll report back here with my findings.

**Comment by nicholas** [comment]
Alright, think I got it narrowed down.

I'm using a library that depends on https://github.com/chrisballinger/FFmpeg-iOS. Problem is, that includes a big set of headers, one of which is [time.h](https://www.ffmpeg.org/doxygen/1.2/time_8h_source.html). Note that it has the same name of the platform provided time header, time.h. So, I'm guessing the FFmpeg time.h gets loaded instead of the one we need. Renaming the FFmpeg time.h to ff_time.h verified that.

In a nutshell, the sqlite amalgamation was calling undeclared time functions.

I think we can safely close this one out - definitely not a Keen issue. :)

**Comment by heitortsergent** [comment]
I'm glad to hear that @nicholas, thanks for digging into this! :)

**Comment by nicholas** [comment]
@heitortsergent no problem. Glad I have a fallback with the binary.

And just to add some more context to this for anyone who stumbles here. A similar issue has been discussed over at [FFmpeg's tracker](https://trac.ffmpeg.org/ticket/1783). Yay name collisions.

**Comment by Koshub** [comment]
In XCode in Header Search Path change "recursive" option to "non-recursive"
